hail/python/hailtop/batch/backend.py

```diff
@@ -51,6 +51,24 @@ class Backend(abc.ABC, Generic[RunningBatchType]):
 
     _closed = False
 
+    def __init__(self):
+        self._batch_fses: Dict["batch.Batch", "RouterAsyncFS"] = {}
```

Comment:
Can we make the key here `GCSRequesterPaysConfiguration`? It avoids (or at least reduces) the circular dependency between `Batch` and `Backend` and allows batches with the same requester pays configuration (which is probably most if not all of them) to reuse the same FS.

Reply:
updated